pkg/app/piped/executor/lambda/lambda.go

```diff
@@ -82,16 +84,65 @@ func decideRevisionName(in *executor.Input, fm provider.FunctionManifest, commit
 	return
 }
 
-func apply(ctx context.Context, in *executor.Input, cloudProviderName string, cloudProviderCfg *config.CloudProviderLambdaConfig, fm provider.FunctionManifest) bool {
+func sync(ctx context.Context, in *executor.Input, cloudProviderName string, cloudProviderCfg *config.CloudProviderLambdaConfig, fm provider.FunctionManifest) bool {
 	in.LogPersister.Infof("Start applying the lambda function manifest")
 	client, err := provider.DefaultRegistry().Client(cloudProviderName, cloudProviderCfg, in.Logger)
 	if err != nil {
 		in.LogPersister.Errorf("Unable to create Lambda client for the provider %s: %v", cloudProviderName, err)
 		return false
 	}
 
-	if err := client.Apply(ctx, fm); err != nil {
-		in.LogPersister.Errorf("Failed to apply the lambda function manifest (%v)", err)
+	ok, err := client.AvailableFunctionName(ctx, fm.Spec.Name)
+	if err != nil {
+		in.LogPersister.Errorf("Unable to validate function name %s: %v", fm.Spec.Name, err)
+		return false
+	}
+	if ok {
+		if err := client.CreateFunction(ctx, fm); err != nil {
+			in.LogPersister.Errorf("Failed to create lambda function %s: %v", fm.Spec.Name, err)
+			return false
+		}
+	} else {
+		if err := client.UpdateFunction(ctx, fm); err != nil {
+			in.LogPersister.Errorf("Failed to update lambda function %s: %v", fm.Spec.Name, err)
+			return false
+		}
+	}
+
+	// Wait before ready to commit change.
+	in.LogPersister.Info("Waiting to update lambda function in progress...")
+	time.Sleep(3 * time.Minute)
```

Comment:
fyi, the Lambda's `CreateFunction/UpdateFunction` API call returns right after we make a request to update the Lambda function code/config, but the update process still running internally in Lambda servers. If we make `PublishFunction` request immediately, the error `ResourceConflictException: The operation cannot be performed at this time.` will be raised.

Reply:
Really? 😢  I think fixed 3 minutes is not safe for all cases, but we don't know how much is good.
So I think in addition to waiting for a fixed amount of time (e.g. 1m) we should also do `PublishFunction` with a constant backoff. 
https://github.com/pipe-cd/pipe/tree/master/pkg/backoff

Reply:
/ping

Reply:
I'm working on it, but shall we make another PR to fix this separately from this PR 👀

Reply:
Yes, that is ok. Let's add a TODO and do it in another PR.